kong/clustering/compat/checkers.lua

```diff
@@ -23,6 +23,70 @@ end
 
 
 local compatible_checkers = {
+  { 3005000000, --[[ 3.5.0.0 ]]
+    function(config_table, dp_version, log_suffix)
+      local has_update
+
+      for _, plugin in ipairs(config_table.plugins or {}) do
+        if plugin.name == 'opentelemetry' or plugin.name == 'zipkin' then
+          local config = plugin.config
+          if config.header_type == 'gcp' then
+            config.header_type = 'preserve'
+            log_warn_message('contains configuration header_type == gcp',
+                              'overwritten with default value `preserve`',
+                              dp_version, log_suffix)
+            has_update = true
+          end
+        end
+
+        if plugin.name == 'zipkin' then
+          local config = plugin.config
+          if config.default_header_type == 'gcp' then
+            config.default_header_type = 'b3'
+            log_warn_message('contains configuration header_type == gcp',
+                              'overwritten with default value `b3`',
+                              dp_version, log_suffix)
+            has_update = true
+          end
+        end
+      end
+
+      return has_update
+    end,
+  },
+
+  { 3004000000, --[[ 3.4.0.0 ]]
+    function(config_table, dp_version, log_suffix)
+      local has_update
+
+      for _, plugin in ipairs(config_table.plugins or {}) do
+        if plugin.name == 'opentelemetry' or plugin.name == 'zipkin' then
+          local config = plugin.config
+          if config.header_type == 'aws' then
+            config.header_type = 'preserve'
+            log_warn_message('contains configuration header_type == aws',
+                              'overwritten with default value `preserve`',
+                              dp_version, log_suffix)
+            has_update = true
+          end
+        end
+
+        if plugin.name == 'zipkin' then
+          local config = plugin.config
+          if config.default_header_type == 'aws' then
+            config.default_header_type = 'b3'
+            log_warn_message('contains configuration header_type == aws',
```

Comment:
s/header_type/default_header_type/

```suggestion
            log_warn_message('contains configuration default_header_type == aws',
```